File: CS/CustomCellColoring/Model/MyCategory.cs
using DevExpress.Mvvm.POCO;
using System.Windows.Media;

namespace CustomCellColoring.Model
{
    public class MyCategory
        {
            protected MyCategory(int id, string name, Color color)
            {
                Id = id;
                Name = name;
                ColorInfo = color;
            }
            public static MyCategory Create(int id, string name, Color color)
            {
                return ViewModelSource.Create(() => new MyCategory(id, name, color));
            }

            protected MyCategory() { }

            public virtual int Id { get; set; }
            public virtual string Name { get; set; }
            public virtual Color ColorInfo { get; set; }
        }
}

File: VB/CustomCellColoring/Model/MyCategory.vb
Imports DevExpress.Mvvm.POCO
Imports System.Windows.Media

Namespace CustomCellColoring.Model

    Public Class MyCategory

        Protected Sub New(ByVal id As Integer, ByVal name As String, ByVal color As Color)
            Me.Id = id
            Me.Name = name
            ColorInfo = color
        End Sub

        Public Shared Function Create(ByVal id As Integer, ByVal name As String, ByVal color As Color) As MyCategory
            Return ViewModelSource.Create(Function() New MyCategory(id, name, color))
        End Function

        Protected Sub New()
        End Sub

        Public Overridable Property Id As Integer

        Public Overridable Property Name As String

        Public Overridable Property ColorInfo As Color
    End Class
End Namespace
